src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1386,6 +1386,21 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		if (!type_enum_is_float (arg0_type))
 			return NULL;
 		return emit_simd_ins_for_sig (cfg, klass, OP_XOP_OVR_X_X, INTRINS_AARCH64_ADV_SIMD_FSQRT, arg0_type, fsig, args);
+#elif TARGET_AMD64
+		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
+		int size = mono_class_value_size (arg_class, NULL);
+		
+		if ( size != 16 ) 		// Only works with Vector128
+			return NULL;
+
+		if (!is_element_type_primitive (fsig->params [0]))
+			return NULL;
```

Comment:
These checks can be moved outside the ifdefs.

Reply:
I could move the `		if (!is_element_type_primitive (fsig->params [0]))
			return NULL;` check outside the ifdef but the size check should stay inside the `#elif TARGET_AMD64` since on ARM64 we support Vector64, right?

Reply:
Yes